### ISSUE TITLE
Console input broken after killing a process in a console where Atuin was called

### ISSUE THREAD

**Issue by yassendobrev** [atuinsh/atuin#492]
Steps to reproduce issue:
In console 1:
1. Call `atuin` with Ctrl+R. It doesn't matter what action is performed in `atuin` (close with `Esc` or choose a command from the history).
2. After closing `atuin` with `Esc`, start a process that runs over a longer time: For example `sleep 100`.

In console 2:
1. Kill the process which was started in console 1: `pkill sleep`.

Return to console 1:
After this, when input is entered in this console, it is not printed anymore, but it gets executed when `Enter` is pressed. This persists until this console is closed. Alternatively, the command `reset` or `tset` also fixes the console.

I have experienced this on bash on Ubuntu 20.04 and Raspberry Pi OS 11.

**Comment by conradludgate** [comment]
I can't recreate using zsh (on MacOS). Can you confirm that if you repeat the steps without atuin no longer happens??

**Comment by yassendobrev** [comment]
Yes, I confirm that this doesn't happen unless I call atuin in console 1.

**Comment by jmfGithub** [comment]
Hi,
Same problem, I can recover the "echo" in console with tset

**Comment by conradludgate** [comment]
Ok, finally got around to looking into this. It definitely exists on latest bash on MacOS. I'll see what I can find

**Comment by felipesere** [comment]
I was unable to reproduce this when running bash within zsh (i.e. just type bash) and tmux.

**Comment by matbgn** [comment]
Having the same issue on Ubuntu 22.04

**Comment by patricksjackson** [comment]
I don't fully understand why, but by removing the `tput rmkx` and `tput smkx` commands from the bash history function I can no longer repro: https://github.com/ellie/atuin/blob/main/src/shell/atuin.bash#L19-L21

I don't _think_ those are actually needed. I've tested in `alacritty`, `foot`, and `st` on linux (so advanced and simple terminals) and the interactive search UI seems to still work fine.

Zsh has equivalent commands as well but they don't seem to cause problems.

Bash may be broken because of when the preexec functions are executed. When you invoke the history command with ^R and then run another command, the preexec function is executed before the history command and not before the execution of the following command. So the next command isn't recorded correctly in atuin. I'm not sure if this is fixable, we'd need to ask https://github.com/rcaloras/bash-preexec

I added print statements to the precmd/preexec and history functions to try to show what I'm talking about.

```
~/tmp
❯ bash
Entering _atuin_precmd

[patrick@carbon:~/tmp]$ ls
Entering _atuin_preexec: ls
file.txt
Entering _atuin_precmd

[patrick@carbon:~/tmp]$ echo "recorded twice"
Entering _atuin_preexec: echo "recorded twice"
recorded twice
Entering _atuin_precmd

Entering _atuin_preexec: echo "recorded twice"
Entering __atuin_history
[patrick@carbon:~/tmp]$ echo "never recorded"
never recorded
Entering _atuin_precmd

[patrick@carbon:~/tmp]$ ls
Entering _atuin_preexec: ls
file.txt
Entering _atuin_precmd

[patrick@carbon:~/tmp]$ atuin history list | tail --lines 6
Entering _atuin_preexec: atuin history list | tail --lines 6
2022-11-17 19:21:55     bash    65ms
2022-11-17 19:21:58     ls      36ms
2022-11-17 19:22:02     echo "recorded twice"   31ms
2022-11-17 19:22:03     echo "recorded twice"   5s
2022-11-17 19:22:14     ls      22ms
2022-11-17 19:22:28     atuin history list | tail --lines 6     0s
Entering _atuin_precmd

[patrick@carbon:~/tmp]$
```

**Comment by ljrk0** [comment]
@patricksjackson Seems you're right.  I don't really want to go into the depths of bash, but simply overriding the function in `.bashrc`:

```
[[ -f ~/.bash-preexec.sh ]] && source ~/.bash-preexec.sh
eval "$(atuin init bash)"
# Override, remove tput statements
__atuin_history ()
{
    HISTORY="$(RUST_LOG=error atuin search -i -- "${READLINE_LINE}" 3>&1 1>&2 2>&3)"

    READLINE_LINE=${HISTORY}
    READLINE_POINT=${#READLINE_LINE}
}
```

Is a workable workaround for me.

**Comment by darthShadow** [comment]
Probably similar to this: https://github.com/cantino/mcfly/issues/247

Upstream report: https://lists.gnu.org/archive/html/bug-bash/2022-03/msg00019.html

Which was fixed in bash version 5.2.0.

**Comment by ellie** [comment]
Resolved by #1170 🙌

**Comment by adrianbienias** [comment]
> Resolved by #1170 🙌

I'm having this issue in `Atuin 18.2.0` with `Bash 5.1.16` on `Ubuntu 22.04.4`
